fvm/src/kernel/default.rs

```diff
@@ -275,11 +275,11 @@ where
         Ok((id, stat))
     }
 
-    fn block_create(&mut self, codec: u64, data: &[u8]) -> Result<BlockId> {
+    fn block_create(&mut self, codec: u64, data: &(impl AsRef<[u8]> + ?Sized)) -> Result<BlockId> {
         self.call_manager
-            .charge_gas(self.call_manager.price_list().on_block_create(data.len()))?;
+            .charge_gas(self.call_manager.price_list().on_block_create( core::mem::size_of_val(data.as_ref())))?;
```

Comment:
`len` is still the correct thing to call here. _Technically_, `size_of_val` will work, but it has different semantics (byte-size of an arbitrary dynamically sized type, versus the length of a byte buffer).

> core

Given that we depend on `std`, just use `std`. Use `core` in `no_std` crates.